Azure-Samples/aks-store-demo
# Refactor for azure

## Purpose

This branch introduces significant changes to the order-service and makeline-service projects so that RabbitMQ and MongoDB can be swapped out for Azure Service Bus and Azure CosmosDB respectively. 

Since Azure Service Bus uses AMQP 1.0, the underlying libs for order-service and makeline-service has been swapped out with with ones that support this protocol. Also, local instances of RabbitMQ will have the amqp1_0 plugin enabled.

This will resolve issues #42 and #43.

## Does this introduce a breaking change?

```
[x] Yes, but only environment variable names
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[x] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone https://github.com/azure-samples/aks-store-demo
cd aks-store-demo
git checkout refactor-for-azure
```

* Run the code using docker compose

```
docker compose up --build
```

* You can also run the code using KIND

```
make all
```

* Test the code

You should see docker compose logs that shows orders being submitted by virtual-customer and processed by the virtual-worker.

Also, using a web browser, navigate to the store-front and store-admin sites.

> NOTE: If using a KIND cluster, you will need to port-forward the store-front and store-admin services using the command below.

```
kubectl port-forward svc/store-front 8080:80
kubectl port-forward svc/store-admin 8081:80
```

## What to Check
Verify that the following are valid
* Ensure the store-front can submit orders and orders can be processed from store-admin
* Ensure the ai-service generates product descriptions from OpenAI

## Other Information
